src/main/java/com/esri/samples/hydrography/add_enc_exchange_set/AddEncExchangeSetSample.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2019 Esri.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.hydrography.add_enc_exchange_set;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.GeometryEngine;
+import com.esri.arcgisruntime.hydrography.EncCell;
+import com.esri.arcgisruntime.hydrography.EncDataset;
+import com.esri.arcgisruntime.hydrography.EncExchangeSet;
+import com.esri.arcgisruntime.layers.EncLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class AddEncExchangeSetSample extends Application {
+
+  private MapView mapView;
+  private Envelope completeExtent;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // set the title and width of the stage
+      stage.setTitle("Add ENC Exchange Set Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+
+      // create a stack pane and set it as the JavaFX scene's root
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map view and add it to the stack pane
+      mapView = new MapView();
+      stackPane.getChildren().add(mapView);
+
+      // create a map with the Oceans basemap and set it on the map view
+      ArcGISMap map = new ArcGISMap(Basemap.createOceans());
+      mapView.setMap(map);
+
```

Comment:
Add a draw status changed listener: the ENC data takes a while to render and from a user perspective it would be useful to have a loading bar/label to show something is happening. e.g.

     ` mapView.addDrawStatusChangedListener(e -> {
        if (e.getDrawStatus() == DrawStatus.IN_PROGRESS) {
          progressBar.setProgress(-100.0);
        } else if (e.getDrawStatus() == DrawStatus.COMPLETED) {
          progressBar.setVisible(false);
        }
        });`